### PR TITLE
Add download_from_url service function

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

1. Rename web_search dir into web
2. Add download_from_url service function
3. Fix import error in distributed search example

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: src/agentscope/service/web/download.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        with open(filepath, "wb") as file:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                progress_bar.update(len(chunk))
+                file.write(chunk)
+        progress_bar.close()
+
+        return ServiceResponse(
+            status=ServiceExecStatus.SUCCESS,
+            content=f"Downloaded file from {url} to {filepath}",
```
Comment (ZiTao-Li) [inline src/agentscope/service/web/download.py:59]: Is it better for LLM to understand/use as a tool if the content is in dict/json format? E.g., {"url": url, "saved_file_path": filepath}

**Review by ZiTao-Li — COMMENTED** [review]
Is it possible to have a unit test to cover the download service?

File: src/agentscope/service/web/download.py (hunk 1 above)
Comment (DavdGao) [reply src/agentscope/service/web/download.py]: Done

File: src/agentscope/service/web/arxiv.py
Comment (pan-x-c) [inline src/agentscope/service/web/arxiv.py:1]: rename `web_search` to `web` in `docs/sphinx_doc/zh_CN/source/agentscope.service.rst` and `docs/sphinx_doc/en/source/agentscope.service.rst`

**Review by ZiTao-Li — APPROVED** [review]
LGTM

**Pull request merged**